### PR TITLE
WIP: Stuff you didn't ask for 2: electric bugaloo (Docker build + CI)

### PR DESCRIPTION
**Pull request by Eriner**
Not quite finished, but mostly there.

* Added CI for testing, linting, and building the Docker image.
* Added `CODEOWNERS` so you will be pinged to review any changes to the `.github/workflows` folder, which is security sensitive. Also added myself for the `Dockerfile`.
* Moved the `golangci-lint` options out of the `Makefile` and into `.golangci-lint.yml` so that the CI tooling (and my editor) will pick up the options. Also added some ignores.
* Fixed misc linting issues.

The Docker images will be pushed to your (@lukechampine's) GitHub package registry: https://github.com/lukechampine?tab=packages

GitHub has made this is \*free (for now).

TBD:
  - [ ] Docker image tags (currently just builds HEAD -> `latest`)

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Dockerfile @eriner
+
+### Security Sensitive ###
+.github/workflows/ @lukechampine
```
Comment (lukechampine) [inline .github/CODEOWNERS]: I don't think a CODEOWNERS file is necessary for a project of this size. I will review every pull request.

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    branches: [master]
+    tags:
+      - v*.*.*
+    # If the Docker build below wasn't a "FROM scratch" container, we would
+    # want to re-build daily to ensure dependencies in the container are
+    # upgraded periodically, even if the app code does not change.
+    # ...But this is a "FROM scratch" container, so we only really have
+    # to worry about Go (the toolchain) vulnerabilities. Meh?
+    #schedule:
+    #  - cron: '0 0 * * *'
```
Comment (lukechampine) [inline .github/workflows/master.yml:14]: my preference would be to never ship automatically-upgraded builds

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    - funlen
+    - gocognit
+    - godox
+    - wsl
+    - exhaustivestruct
+    - nlreturn
+    - wrapcheck
+    - prealloc
+    - goerr113 # fixing this (returning wrapped static error types) should be a TODO at some point, not worrying about it now.
+    - cyclop
```
Comment (lukechampine) [inline .golangci.yml]: I no longer use golangci-lint in my projects. I don't think it adds enough value to be worth it. Let's strip it out (and revert the linting changes).

File: client.go [hunk 4]
```diff
@@ -503,5 +509,5 @@ func (c *protoBridge) UnconfirmedParents(txn types.Transaction) ([]types.Transac
 
 func (c *protoBridge) FeeEstimate() (minFee, maxFee types.Currency, err error) {
 	fee, err := c.Client.RecommendedFee()
-	return fee, fee.Mul64(3), err
+	return fee, fee.Mul64(3), err //nolint:gomnd
```
Comment (lukechampine) [inline client.go]: lol. This is exactly the sort of thing I don't like to see.

File: cmd/walrus/main.go [hunk 5]
```diff
@@ -16,7 +16,7 @@ import (
 )
 
 var (
-	// to be supplied at build time
+	// to be supplied at build time.
```
Comment (lukechampine) [inline cmd/walrus/main.go]: my (admittedly idiosyncratic) commenting style is "lowercase without period for up to 1 sentence, proper grammar for anything longer." Not going to change that to suit some nonstandard linting option.

**Review by lukechampine — COMMENTED** [review]
I'm willing to trial this for *one* repository (of `walrus`, `shard`, `muse`, `user`), so choose carefully. As I see it, they're all pretty stable at this point and aren't under active development, so the value of adding CI is questionable. But I haven't given GitHub Actions a proper try yet, and it seems worth investigating.

**Comment by Eriner** [comment]
Ha, does it really matter which, then? :) I still need to handle the tagging, so still WIP. Comments addressed.

**Comment by Eriner** [comment]
The only reason CI is worth adding is to produce the Docker containers as far as I'm concerned, but I figured I'd make it a bit more useful and add the traditional `make test` and whatnot. (I'm only half kidding)

File: .github/workflows/master.yml (hunk 2 above)
Comment (Eriner) [reply .github/workflows/master.yml:14]: The current implementation will build a new container with the `latest` tag for each push to master. Once I get the tagging sorted out, `latest` will point to the latest tagged release.

Comment (Eriner) [reply .github/workflows/master.yml:14]: I would disagree re: the value of automatic updates/rebuilds if the `Dockerfile` wasn't created the way it is, but the underlying concern (OS packages with vulns in the container) is more an ops problem than a dev problem.

**Pull request merged**